### PR TITLE
[Forums] Initial forums API

### PR DESCRIPTION
**Pull request by aanguyen**
**Summary:**
Extremely basic API that allows creation and retrieval of forum posts, with only minimal properties. There's no validation yet on the actual user posting though.

I generally prefer a workflow with many small commits, but let me know if you feel like there should be more functionality before this gets merged 😛 

**Test plan:**
Unit tests pass, was able to POST and GET successfully, manually verified on Mongo express interface.

### CODE REVIEW

File: packages/backend/src/forums/forums.service.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    const result = await newPost.save();
+    return result.id as string;
+  }
+
+  async getPosts() {
+    const posts = await this.ForumPostModel.find().exec();
+    return posts as ForumPost[];
+  }
+
+  async getSinglePost(id: string) {
```
Comment (vck3000) [inline packages/backend/src/forums/forums.service.ts:25]: Let's rename this to `getPost()`

File: packages/backend/src/forums/forums.service.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  ) {}
+
+  async addPost(createForumpostDto : CreateForumpostDto) {
+    const newPost = new this.ForumPostModel(createForumpostDto);
+    const result = await newPost.save();
+    return result.id as string;
+  }
+
+  async getPosts() {
+    const posts = await this.ForumPostModel.find().exec();
```
[the remark this answers is not in the document]
Comment (vck3000) [reply packages/backend/src/forums/forums.service.ts:21]: We may need to extend on this or add in a new method to get the forum posts for a particular "page".

**Comment by aanguyen** [comment]
Not sure if I did the e2e test correctly but happy to get feedback :v

**Comment by vck3000** [comment]
Great work! Looks awesome now :) Thank you!

**Pull request merged**